clang_format.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+echo "start formatting files ..."
+for file in `find ./ -type f -regex '.*\.[h|m]$'`; do
+    clang-format -i ${file} --verbose || exit 1
```

Comment:
nit: do we want to make sure that `clang_format` exists here? in case someone hasn't called `./bootstrap` yet?

Reply:
I tried once locally, basically the script will exit 1 with message like "clang-format not found". I guess we could really beef up our README, to give instructions there.